src/controller/python/chip-repl.py

```diff
@@ -18,31 +18,17 @@
 #
 
 import IPython
-import chip
-import chip.logging
-import coloredlogs
-import logging
 from traitlets.config import Config
-from rich import print
-from rich import pretty
-from rich import inspect
 import builtins
-import argparse
 import sys
 
 
 def main():
-    parser = argparse.ArgumentParser()
```

Comment:
Does this still achieve the effect of passing in args to this chip-repl script, and being able to get help text for them?

Reply:
Yes, it's just that now you get some of the REPL startup output first before you see the help:

```
Python 3.9.12 (main, Apr 29 2022, 23:24:05) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.3.0 -- An enhanced Interactive Python. Type '?' for help.
[1652476661.760088][1135997:1135997] CHIP:CTL: Setting attestation nonce to random value
[1652476661.760328][1135997:1135997] CHIP:CTL: Setting CSR nonce to random value
[1652476661.763484][1135997:1135997] CHIP:DL: writing settings to file (/tmp/chip_counters.ini-RCpKWc)
[1652476661.763552][1135997:1135997] CHIP:DL: renamed tmp file to file (/tmp/chip_counters.ini)
[1652476661.763561][1135997:1135997] CHIP:DL: NVS set: chip-counters/reboot-count = 73 (0x49)
[1652476661.764178][1135997:1135997] CHIP:DL: Got Ethernet interface: enp6s0
[1652476661.764643][1135997:1135997] CHIP:DL: Found the primary Ethernet interface:enp6s0
[1652476661.765104][1135997:1135997] CHIP:DL: Got WiFi interface: wlp5s0
[1652476661.765113][1135997:1135997] CHIP:DL: Failed to reset WiFi statistic counts
[1652476661.765160][1135997:1136005] CHIP:DL: Platform main loop started.
usage: ChipReplStartup.py [-h] [-p STORAGEPATH] [-d]

optional arguments:
  -h, --help            show this help message and exit
  -p STORAGEPATH, --storagepath STORAGEPATH
                        Path to persistent storage configuration file (default: /tmp/repl-storage.json)
  -d, --debug           Set default logging level to debug.

In [1]: 
```

Just noticed you actually land in the REPL command prompt, with a half way setup CHIP REPL. I check if I can improve that.

Reply:
That seems not to be possible as we are in IPython `%run` which seems to ignore even explicit `sys.exit()` from `ChipReplStartup.py`.

I kinda dislike to duplicate argument parsing, maybe move parsing back to `chip-repl.py` and set some variables via `exec_lines`?